Data/Repositories/RecruitmentRepository.cs

```diff
@@ -1,36 +1,46 @@
+using AutoMapper;
 using Common.Enums;
 using Data.Entities;
 using Microsoft.Data.SqlClient;
+using Data.DTOs.Recruitment;
 using System.Security.Claims;
 
+
 namespace Data.Repositories
 {
     public class RecruitmentRepository : BaseRepository<Recruitment>
     {
-        public RecruitmentRepository(DataContext context) : base(context)
+        private readonly IMapper _mapper;
+        public RecruitmentRepository(DataContext context, IMapper mapper) : base(context)
         {
-            
+            _mapper = mapper;
         }
-
-        public int GetRecruiterId(int id)
+        public RecruitmentDetailsDTO GetRecruitmentById(int id) //RecruitmentDetailsDTO
         {
             Recruitment recruitment = GetById(id);
-            int result = recruitment.RecruiterId;
+
+            if (recruitment.DeletedDate.HasValue) return null;
```

Comment:
Add new line. what if recruitment is null?

Reply:
Ok, I will do that